crates/evm/core/src/constants.rs

```diff
@@ -47,6 +48,23 @@ pub const DEFAULT_CREATE2_DEPLOYER_CODE: &[u8] = &hex!("604580600e600039806000f3
 /// The runtime code of the default CREATE2 deployer.
 pub const DEFAULT_CREATE2_DEPLOYER_RUNTIME_CODE: &[u8] = &hex!("7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffe03601600081602082378035828234f58015156039578182fd5b8082525050506014600cf3");
 
+/// The lazily initialized CREATE2 deployer address that can be configured via environment
+/// variable or the default one.
+pub static CREATE2_DEPLOYER: LazyLock<Address> = LazyLock::new(|| {
+    match std::env::var("FOUNDRY_CREATE2_DEPLOYER") {
+        Ok(addr) => Address::from_hex(addr).unwrap_or_else(|_| {
+            error!("env FOUNDRY_CREATE2_DEPLOYER is set, but not a valid Ethereum address, use {DEFAULT_CREATE2_DEPLOYER} instead");
+            DEFAULT_CREATE2_DEPLOYER
+        }),
+        Err(_) => DEFAULT_CREATE2_DEPLOYER,
+    }
+});
```

Comment:
@grandizzy wdyt on making this a `Config` field? so that you can configure it through foundry.toml as well

Reply:
good point, yep, would be great to have such